vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet_util.cc

```diff
@@ -186,11 +186,17 @@ type::ExternalWalletPtr ResetWallet(type::ExternalWalletPtr wallet,
   wallet = type::ExternalWallet::New();
   wallet->type = wallet_type;
 
-  if (status != type::WalletStatus::NOT_CONNECTED) {
+  if (wallet_type == constant::kWalletUphold) {
```

Comment:
It seems like with this change, the logic here should work for all wallet types, since there's no `CONNECTED` state anymore. If we need to keep this branch around, we should probably add a comment explaining that the special case should be removed after we're sure it's OK (perhaps a `TODO(...): ...` comment.

Reply:
```
if (status != type::WalletStatus::NOT_CONNECTED) {
  if (status == type::WalletStatus::VERIFIED) {
    wallet->status = type::WalletStatus::DISCONNECTED_VERIFIED;
  } else {
    wallet->status = type::WalletStatus::DISCONNECTED_NOT_VERIFIED;
  }
}
```
doesn't suffice for Uphold, since
```
NOT_CONNECTED         ==> NOT_CONNECTED             (OK)
VERIFIED              ==> DISCONNECTED_VERIFIED     (OK)
DISCONNECTED_VERIFIED ==> DISCONNECTED_NOT_VERIFIED (not OK, should be NOT_CONNECTED)
PENDING               ==> DISCONNECTED_NOT_VERIFIED (not OK, should be NOT_CONNECTED)
```

Reply:
Right. Would the "uphold" branch work now for all external wallet providers, though?

```cpp
if (status == type::WalletStatus::VERIFIED) {
  wallet->status = type::WalletStatus::DISCONNECTED_VERIFIED;
}
```

since the others don't support `CONNECTED` anyway?

Reply:
It should work for all custodians regardless of whether they support it or not, since we check if we're resetting the wallet for Uphold:
```
if (wallet_type == constant::kWalletUphold) {
// ...
}
```
AFAIR, the `bitFlyer` implementation never assigns `CONNECTED` to `status` by the way, `Gemini` supports it.